testplan/runners/pools/base.py

```diff
@@ -259,25 +292,27 @@ def __init__(
         worker_type=Worker,
         worker_heartbeat=None,
         heartbeats_miss_limit=3,
-        task_retries_limit=3,
-        max_active_loop_sleep=5,
         restart_count=3,
+        max_active_loop_sleep=5,
+        should_rerun=default_check_rerun,
         **options
     ):
         options.update(self.filter_locals(locals()))
         super(Pool, self).__init__(**options)
         self.unassigned = []  # unassigned tasks
-        self.task_assign_cnt = {}  # uid: times_assigned
-        self.should_reschedule = default_check_reschedule
+        self._task_retries_cnt = {}  # uid: times_reassigned_without_result
+        self._task_retries_limit = 2
+        self._task_rerun_limit = 3
```

Comment:
Let's control rerun limit only at Task. I prefer not having yet another limit in Pool to control the same behavior